aggregate/housing_security/units_affordable.py

```diff
@@ -0,0 +1,86 @@
+from typing import final
+import pandas as pd
+from utils.PUMA_helpers import clean_PUMAs
+from internal_review.set_internal_review_file import set_internal_review_files
+from aggregate.clean_aggregated import order_affordable
+from utils.PUMA_helpers import (
+    community_district_to_PUMA,
+    clean_PUMAs,
+    borough_name_mapper,
+)
+
+ind_mapper = {
+    "Af": "units_affordable_",
+    "ROcc2": "units_renteroccu"
+}
+
+income_mapper = {
+    "ELI": "eli",
+    "VLI": "vli",
+    "LI": "li",
+    "MI": "mi",
+    "Midi": "midi",
+    "HI": "hi",
+}
+
+suffix_mapper = {
+    "_19E": "",
+    "_19M": "_moe",
+    "_19C": "_cv",
+    "_19P": "_pct",
+    "_19Z": "_pct_moe",
+}
+
+def units_affordable(
+    geography: str, write_to_internal_review=False
+    ) -> pd.DataFrame:
+    assert geography in ["citywide", "borough", "puma"]
+
+    clean_df = load_source_clean_data()
+    if geography == "puma":
+        final = clean_df.loc[clean_df.Geog.str[0].isna()].copy()
+        final["puma"] = final.Geog.apply(func=clean_PUMAs)
+    elif geography == "borough":
+        clean_df["borough"] = clean_df.Geog.map(borough_name_mapper)
+        final = clean_df.loc[~clean_df.borough.isna()].copy()
+    else:
+        clean_df.loc[clean_df.Geog == "NYC", "citywide"] = "citywide"
+        final = clean_df.loc[~clean_df.citywide.isna()].copy()
```

Comment:
is there a reason to take a copy here?

Reply:
I think this is only to avoid the warning that otherwise come from .loc for assigning to a copy of dataframe